### ISSUE TITLE
[Bug]: `enforce_theme=default` Disables Browser-Based Dark/Light Mode Detection

### ISSUE THREAD

**Issue by bromiesTM** [nextcloud/server#46217]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When the `enforce_theme` parameter in `config.php` is set to `default`, Nextcloud does not automatically adjust the theme (dark mode or light mode) based on the browser's settings.

### Steps to reproduce

1. Set the `enforce_theme` parameter in `config.php` to `default`:
2. Open Nextcloud in Browser and confirm default theme is enforced at settings/user/theming
3. Observe that Nextcloud does not change its theme based on the browser's theme settings. (Dev Tools ⇾ Rendering ⇾ Emulate CSS media feature prefers-color-scheme)

### Expected behavior

Nextcloud should automatically switch between dark mode and light mode based on the browser's theme settings when the enforce_theme parameter is set to default.

### Installation method

None

### Nextcloud Server version

29

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by kesselb** [comment]
cc @susnux @skjnldsv

**Comment by skjnldsv** [comment]
Ah, we should just remove `default` from the example in config.sample.php.
Default is not a valid theme, enforcing it makes zero sense :)

**Comment by pandusen** [comment]
> Ah, we should just remove `default` from the example in config.sample.php. Default is not a valid theme, enforcing it makes zero sense :)
Currently it does make sense. It is the only way to force browser based choice of theme. for me, that makes sense, as I am using a custom theme, that is designed to work in dark and light mode.

If I force either dark or light, it will overwrite the theme completely.
If I dont set any value, the users will be able to change the theme.

**Comment by skjnldsv** [comment]
`custom theme` as within the `themes` directory of your Nextcloud root?

**Comment by pandusen** [comment]
> `custom theme` as within the `themes` directory of your Nextcloud root?
<

Correct. It used to be that you could just write your theme name "my_theme" directly in "enforce_theme" to apply the theme, and disable user theming. But as always with nextcloud, admin control is slowly getting deprecated.. Now you can only use the build-in themes: dark, dark-highcontrast, default, light, light-highcontrast, opendyslexic. Where default is the current theme.

**Comment by bromiesTM** [comment]
Exactly, that's why enforcing `default` should prevent the non-privileged user from switching themes manually while preserving the auto-adjusting functionality based on browser / OS settings.
At least that's what I expected.

**Comment by skjnldsv** [comment]
@pandusen yeah, the custom theming feature you're using is slowly getting deprecated for a few years now.
I would no rely on it on the long run, sorry :)
